crates/bitwarden-auth/src/identity/login_via_password/password_prelogin_response.rs

```diff
@@ -0,0 +1,290 @@
+use std::num::NonZeroU32;
+
+use bitwarden_api_identity::models::{KdfType, PasswordPreloginResponseModel};
+use bitwarden_core::{MissingFieldError, require};
+use bitwarden_crypto::{
+    Kdf, default_argon2_iterations, default_argon2_memory, default_argon2_parallelism,
+    default_pbkdf2_iterations,
+};
+use serde::{Deserialize, Serialize};
+
+/// Response containing the data required before password-based authentication
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))] // add mobile support
+#[cfg_attr(
+    feature = "wasm",
+    derive(tsify::Tsify),
+    tsify(into_wasm_abi, from_wasm_abi)
+)] // add wasm support
+pub struct PasswordPreloginResponse {
+    /// The Key Derivation Function (KDF) configuration for the user
+    pub kdf: Kdf,
+
+    /// The salt used in the KDF process
+    pub salt: String,
```

Comment:
Thought (not for this PR): we should make salt a newtype, since passing it around like this feels dangerous. You could put in a non-lower-cased email and think it's fine, but it would be wrong.